backend/src/db/cache_storage.py

```diff
@@ -34,28 +38,187 @@ def __init__(self):
+
+        Args:
+            pattern (bytes | str | memoryview): The channel to publish the signal to.
+            serialized_data (bytes | memoryview | str | int | float): The data to publish.
+
+        Returns:
+            Awaitable
+        """
+        return self.cache.publish(channel=pattern, message=serialized_data)
+
+    def subscribe_signal(
+        self,
+        pattern: list,
+    ) -> None:
+        """
+        Subscribe to one or more channels to receive signals.
+
+        Args:
+            pattern (list): List of channels to subscribe to.
+
+        Returns:
+            None
+        """
+        self.pubsub.subscribe(*pattern)
+
+    def unsubscribe_signal(
+        self,
+        pattern: list,
+    ) -> None:
+        """
+        Unsubscribe from one or more channels.
+
+        Args:
+            pattern (list): List of channels to unsubscribe from.
+
+        Returns:
+            None
+        """
+        self.pubsub.unsubscribe(*pattern)
+
+    def get_signal(self) -> Optional[Union[None, dict[str, Union[None, str, bytes]]]]:
+        """
+        Get the next signal message from the subscribed channels.
+
+        Returns:
+            Optional[Union[None, dict[str, Union[None, str, bytes]]]]
+        """
+        while message := self.pubsub.get_message():
+            if message and message["type"] == "message":
+                return message
+            time.sleep(0.2)
+        return None
+
+
+class CacheHandler(ICacheHandler):
+    def __init__(self, redis: RedisStorage):
+        """
+        Initialize the CacheHandler with the provided RedisStorage instance.
+
+        Args:
+            redis (RedisStorage): The RedisStorage instance to use for caching.
+        """
+        self.redis = redis
+
+    def publish_event(self, pattern: str, data: dict) -> None:
+        """
+        Publish an event with the specified pattern and data.
+
+        Args:
+            pattern (str): The pattern to publish the event under.
+            data (dict): The data to be published.
+
+        Returns:
+            None
+        """
+        serialized_data = json.dumps(data)
+        self.redis.publish_signal(
+            pattern=pattern,
+            serialized_data=serialized_data,
+        )
+        logger.info("Published event: %s" % data)
+
+    def subscribe_event(self, pattern: str | list[str]) -> None:
+        """
+        Subscribe to events with the specified pattern(s).
+
+        Args:
+            pattern (str | list[str]): The pattern(s) to subscribe to.
+
+        Returns:
+            None
+        """
+        if isinstance(pattern, str):
+            pattern = [pattern]
 
-    def subscribe(self, channel, callback):
-        pass
+            self.redis.subscribe_signal(pattern=pattern)
 
-    def unsubscribe(self, channel):
-        pass
+        logger.info("Subscribed to events: %s" % pattern)
+
+    def unsubscribe_event(self, pattern: str | list[str]) -> None:
```

Comment:
Same as here https://github.com/bh-coders/trip-planer/pull/109/files#r1510380092

Reply:
https://github.com/bh-coders/trip-planer/pull/109#discussion_r1510380330

Reply:
this links not working this way for me ;/

Reply:
https://github.com/bh-coders/trip-planer/pull/109#discussion_r1510380092